<a href="https://colab.research.google.com/github/BernardOforiBoateng/Data_Wrangling_DSCI401/blob/main/DSCI401_HW1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## DSCI401 - Homework 1

# Question 3

### Importing the required library

In [ ]:
import pandas as pd

### Loading the dataset

In [ ]:
cubs = pd.read_csv("https://raw.githubusercontent.com/gjm112/DSCI401/main/data/cubs_all_time.csv")

### Getting a sense of the dataset

In [ ]:
# Loading the first 5 rows
cubs.head()

Year            Tm          Lg    G   W   L  Ties   W-L%  pythW-L%  \
0  2022  Chicago Cubs  NL Central  125  54  71     0  0.432     0.436   
1  2021  Chicago Cubs  NL Central  162  71  91     0  0.438     0.421   
2  2020  Chicago Cubs  NL Central   60  34  26     0  0.567     0.545   
3  2019  Chicago Cubs  NL Central  162  84  78     0  0.519     0.558   
4  2018  Chicago Cubs  NL Central  163  95  68     0  0.583     0.575   

     Finish  ...         Playoffs    R   RA  Attendance  BatAge  PAge  #Bat  \
0  3rd of 5  ...              NaN  518  596   2163550.0    28.1  29.8    57   
1  4th of 5  ...              NaN  705  839   1978934.0    29.1  29.4    69   
2  1st of 5  ...  Lost NLWC (2-0)  265  240         NaN    27.9  30.3    47   
3  3rd of 5  ...              NaN  814  717   3094865.0    27.7  31.1    52   
4  2nd of 5  ...  Lost NLWC (1-0)  761  645   3181089.0    27.2  30.2    50   

   #P         Top Player          Managers  
0  37    N.Hoerner (4.2)    D.Ross (54-71)  
1  40  W.Contreras (4.1)    D.Ross (71-91)  
2  26    Y.Darvish (2.8)    D.Ross (34-26)  
3  33       J.Baez (6.6)  J.Maddon (84-78)  
4  35       J.Baez (6.4)  J.Maddon (95-68)  

[5 rows x 21 columns]

In [ ]:
# Getting the shape of the dataframe
cubs.shape

(147, 21)

# (a)


### Total number of games won between 1876 and 2022

In [ ]:
total_games_won = cubs['W'].sum()
print("Number of games won between 1876 and 2022 by the Cubs:", total_games_won)

Number of games won between 1876 and 2022 by the Cubs: 11141


### Total number of games lost between 1876 and 2022

In [ ]:
total_games_lost = cubs['L'].sum()
print("Number of games lost between 1876 and 2022 by the Cubs:", total_games_lost)

Number of games lost between 1876 and 2022 by the Cubs: 10592


# (b)

### Year the cubs scored the most runs

In [ ]:
#Find the maximum runs scored
most_runs = cubs['R'].max()

# find the year corresponding to the maximum runs
most_runs_year = cubs.loc[cubs['R'] == most_runs, 'Year'].values[0]

print("Year the Cubs scored the most runs:",  most_runs_year)

Year the Cubs scored the most runs: 1894


### Year the cubs scored the fewest runs

In [ ]:
#Find the fewest runs scored
fewest_runs = cubs['R'].min()

# find the year corresponding to the fewest runs
fewest_runs_year = cubs.loc[cubs['R'] == fewest_runs, 'Year'].values[0]

print("Year the Cubs scored the fewest runs:",  fewest_runs_year)

Year the Cubs scored the fewest runs: 2020


### Scatter plot showing the runs scored over the years

In [ ]:
import plotly.express as px

# Scatter plot for runs scored
runs_scored_fig = px.scatter(cubs, x='Year', y='R', title='Cubs Runs Scored Over the Years',
                 labels={'Year': 'Year', 'R': 'Runs Scored'})

runs_scored_fig.update_traces(mode='lines+markers')

runs_scored_fig.show()

### Games played by the Cubs in 2020

In [ ]:
games_played_2020 = cubs.loc[cubs['Year'] == 2020, 'G'].values[0]
print("Number of games played in 2020:", games_played_2020)

Number of games played in 2020: 60


Typically, teams in the MLB play 162 games in the regular season. However, in 2020, during the height of COVID19 pandemic, the MLB made a ddecision to reduce the number of games in the regular season to only 60 games to ensure player, fan and staff safety.
The reduced schedule meant that there were fewer opportunities for the Cubs to accrue runs, resulting in the lowest run count in the Cubs history.

# (c)

### Number of seasons with over 3 million attendance

In [ ]:
attendance_over_3m = (cubs['Attendance'] > 3000000).sum()
print("Number of seasons the Cubs total attendance was over 3 million:", attendance_over_3m)

Number of seasons the Cubs total attendance was over 3 million: 12


### Scatter plot showing the Attendance over the years

In [ ]:
import plotly.express as px

# Scatter plot for runs scored
Attendance_fig = px.scatter(cubs, x='Year', y='Attendance', title='Cubs Attendance Over the Years',
                 labels={'Year': 'Year', 'Attendance': 'Attendance'})

Attendance_fig.update_traces(mode='lines+markers')

Attendance_fig.show()